src/test/java/picard/sam/DownsampleSamTest.java

```diff
@@ -148,5 +151,52 @@ private void testDownsampleWorker(final File samFile, final double fraction, fin
             TestNGUtil.assertGreaterThan(SamTestUtil.countSamTotalRecord(downsampled), fraction * .8 * SamTestUtil.countSamTotalRecord(samFile));
             TestNGUtil.assertLessThan(SamTestUtil.countSamTotalRecord(downsampled), fraction * 1.2 * SamTestUtil.countSamTotalRecord(samFile));
         }
+        return downsampled;
+    }
+
+
+
+    @DataProvider(name = "RepeatedDownsamplingProvider")
+    public Object[][] repeatedDownsamplingProvider() {
+        final List<Object[]> rets = new ArrayList<>();
+        rets.add(new Object[]{Arrays.asList(DownsamplingIteratorFactory.Strategy.ConstantMemory, ConstantMemory), Arrays.asList(2,1)});
+        rets.add(new Object[]{Arrays.asList(ConstantMemory, ConstantMemory), Arrays.asList(1,1)});
+        rets.add(new Object[]{Arrays.asList(Chained, ConstantMemory), Arrays.asList(1,1)});
+        rets.add(new Object[]{Arrays.asList(Chained, ConstantMemory), Arrays.asList(1,3)});
+        rets.add(new Object[]{Arrays.asList(ConstantMemory, Chained), Arrays.asList(1,1)});
+        rets.add(new Object[]{Arrays.asList(HighAccuracy, ConstantMemory), Arrays.asList(1,1)});
+        rets.add(new Object[]{Arrays.asList(ConstantMemory, HighAccuracy), Arrays.asList(1,1)});
+        rets.add(new Object[]{Arrays.asList(Chained, Chained), Arrays.asList(1,1)});
+        rets.add(new Object[]{Arrays.asList(HighAccuracy, Chained), Arrays.asList(1,1)});
+        rets.add(new Object[]{Arrays.asList(HighAccuracy, HighAccuracy), Arrays.asList(1,1)});
+        rets.add(new Object[]{Arrays.asList(Chained, HighAccuracy), Arrays.asList(1,1)});
+
+        //randomly generate some sequences to test out
+        final Strategy[] availableStratagies = Strategy.values();
+        final Random random = new Random(12345);
+        for (int i =0; i<20; i++) {
+            final List<Strategy> strategies = new ArrayList<>();
+            final List<Integer> seeds = new ArrayList<>();
+
+            while (strategies.size() < 5) {
+                final int seed = random.nextInt(3);
+                final Strategy strategy = availableStratagies[random.nextInt(availableStratagies.length)];
+
+                seeds.add(seed);
+                strategies.add(strategy);
+            }
+            rets.add(new Object[]{strategies, seeds});
+        }
+
+        return rets.toArray(new Object[0][]);
+    }
+
+    @Test(dataProvider = "RepeatedDownsamplingProvider")
```

Comment:
please add a test that shows that downsampling twice with 50% actually yields 25%.

Reply:
`testDownsampleWorker` does this in some sense, in that it tests that each downsampling step actually downsamples the data as intended.  I have also added code to test the the cumulative downsampling is as expected.

Reply:
👍